src/Microsoft.Azure.WebJobs.Host/Abstractions/IActivitySourceAbstraction.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+#nullable enable
+
+namespace Microsoft.Azure.WebJobs.Host.Abstractions
+{
+    /// <summary>
+    /// Abstraction for ActivitySource to avoid dependency on System.Diagnostics.DiagnosticSource
+    /// </summary>
+    public interface IActivitySourceAbstraction
+    {
+        IActivityAbstraction? StartActivity(string name, ActivityKindAbstraction kind = ActivityKindAbstraction.Server);
```

Comment:
Do we need this to be nullable? Being explicit would have the benefit of callers clearly expressing what kind they are.

Reply:
`StartActivity` may return `null` in case of sampling or if there is no activity source listener. 
Removed Kind.